FEXCore/include/FEXCore/Config/Config.h

```diff
@@ -123,6 +123,12 @@ namespace DefaultValues {
     FEXCore::Config::DefaultValues::enum                                    \
   }
 
+// Get a configuration option as a single-shot.
+#define FEX_CONFIG_OPT_SINGLE(enum)                                    \
```

Comment:
I'm not fan of adding a separate macro for such a small gain. It doesn't seem worth having to look up which macro is the right one each time a config check is needed (since `FEX_CONFIG_OPT` and `FEX_CONFIG_OPT_SINGLE` aren't exactly the most distinguishable names).

The way this macro is implemented also seems likely to produce bad error messages on syntax errors (which are easier to make with e.g. the if/for examples you gave), so it seems preferable to make people stick to the statement-syntax instead at the expense of an extra line of code.

Reply:
A vaguely thought-through alternative might be to get rid of the macros entirely. I a templated type `ConfigFetcher<FEXCore::Config::CONFIG_MULTIBLOCK>` could be implemented that can either be used in variable declarations or in expressions. We'd need a way to map a `CONFIG_` value to its type (pretty easy) and we'd need to move the default initialization to that type's destructor.

Reply:
Submitted in #5050.